x-pack/plugins/fleet/server/services/agents/agentless_agent.ts

```diff
@@ -155,7 +163,112 @@ class AgentlessAgentService {
       }
     );
 
-    logger.debug(`Created an agentless agent ${response}`);
+    logger.debug(`[Agentless API] Created an agentless agent ${response}`);
+    return response;
+  }
+
+  public async deleteAgentlessAgent(agentlessPolicyId: string) {
+    const logger = appContextService.getLogger();
+    const agentlessConfig = appContextService.getConfig()?.agentless;
+    const tlsConfig = new SslConfig(
+      sslSchema.validate({
+        enabled: true,
+        certificate: agentlessConfig?.api?.tls?.certificate,
+        key: agentlessConfig?.api?.tls?.key,
+        certificateAuthorities: agentlessConfig?.api?.tls?.ca,
+      })
+    );
+
```

Comment:
I think we should create a utility function for hte `SslConfig` and the `https.Agent` below